tests/core/v5_1/alexandria/test_alexandria_client.py

```diff
@@ -48,6 +49,63 @@ async def _respond_empty():
                 )
 
 
+@pytest.mark.trio
+async def test_request_ensures_valid_msg_type(bob, alice_alexandria_client):
+    with pytest.raises(TypeError):
+        await alice_alexandria_client._request(
+            bob.node_id,
+            bob.endpoint,
+            PongMessage(PongPayload(enr_seq=1234, advertisement_radius=4321)),
+            PongMessage,
+            request_id=b"\x01\x02",
+        )
+
+
+@pytest.mark.trio
+async def test_send_response_ensures_valid_msg_type(bob, alice_alexandria_client):
+    with pytest.raises(TypeError):
+        await alice_alexandria_client._send_response(
+            bob.node_id,
+            bob.endpoint,
+            PingMessage(PingPayload(enr_seq=1234, advertisement_radius=4321)),
+            request_id=b"\x01\x02",
+        )
+
+
+@pytest.mark.trio
+async def test_feed_talk_requests_ignores_wrongly_typed_msgs(
+    bob, bob_alexandria_client, alice_alexandria_client
+):
+    async with bob_alexandria_client.network.dispatcher.subscribe(
+        TalkRequestMessage
+    ) as sub:
+        async with bob_alexandria_client.subscribe(PongMessage) as client_subscription:
+
+            # Send a PongMessage via TALKREQ.
+            msg = PongMessage(PongPayload(enr_seq=1234, advertisement_radius=4321))
+            await alice_alexandria_client.network.client.send_talk_request(
+                bob.node_id,
+                bob.endpoint,
+                protocol=ALEXANDRIA_PROTOCOL_ID,
+                payload=msg.to_wire_bytes(),
+                request_id=b"\x01\x02",
+            )
+
+            # The message will be received by bob.
+            with trio.fail_after(1):
+                inbound_msg = await sub.receive()
+
+            assert inbound_msg.request_id == b"\x01\x02"
+
+            # But the alexandria client will not deliver it to any subscribers. And it should
+            # log a warning about it.
+            pong_msg = None
+            with trio.move_on_after(0.2):
```

Comment:
You can raise this timeout a little to avoid false positives and use the [`autojump_clock`](https://pytest-trio.readthedocs.io/en/latest/quickstart.html#trio-s-magic-autojump-clock) to avoid having to actually wait for the timeout.